Start migrating functions using postMessageWithAsyncReply from the web content process to the UI process

colelao opened the pull request:
#### 355afe3587aabdc67b9e6f04ddadc7ed31905d97
<pre>
Start migrating functions using postMessageWithAsyncReply from the web content process to the UI process
<a href="https://bugs.webkit.org/show_bug.cgi?id=298731">https://bugs.webkit.org/show_bug.cgi?id=298731</a>
<a href="https://rdar.apple.com/160392755">rdar://160392755</a>

Reviewed by Alex Christensen.

Started moving the first 10 functions using postMessageWithAsyncReply to the UI process.

* Tools/WebKitTestRunner/InjectedBundle/Bindings/TestRunner.idl:
* Tools/WebKitTestRunner/InjectedBundle/TestRunner.cpp:
(WTR::TestRunner::displayAndTrackRepaints): Deleted.
(WTR::TestRunner::clearBackForwardList): Deleted.
(WTR::TestRunner::addChromeInputField): Deleted.
(WTR::TestRunner::removeChromeInputField): Deleted.
(WTR::TestRunner::setTextInChromeInputField): Deleted.
(WTR::TestRunner::selectChromeInputField): Deleted.
(WTR::TestRunner::getSelectedTextInChromeInputField): Deleted.
(WTR::TestRunner::focusWebView): Deleted.
(WTR::TestRunner::setBackingScaleFactor): Deleted.
(WTR::TestRunner::removeAllCookies): Deleted.
* Tools/WebKitTestRunner/InjectedBundle/TestRunner.h:
* Tools/WebKitTestRunner/TestController.cpp:
(WTR::if):
(WTR::CompletionHandler&lt;void):

Canonical link: <a href="https://commits.webkit.org/300064@main">https://commits.webkit.org/300064@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a540b0ce2d3bf2405dceeeb48e37aaa39015d76d

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/121182 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/40878 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/31536 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/127606 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/73263 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/123058 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/41580 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/49457 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/92057 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/61247 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/124134 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/33207 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/108623 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/72733 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/32233 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/26731 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/71195 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/102716 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/26911 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/130454 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/48109 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/36570 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/100656 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/48477 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/104796 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/100560 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25489 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/45964 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/24026 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/44803 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/47967 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/47438 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/50785 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/49122 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/a540b0ce2d3bf2405dceeeb48e37aaa39015d76d. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50614

colelao committed:
Start migrating functions using postMessageWithAsyncReply from the web content process to the UI process
https://bugs.webkit.org/show_bug.cgi?id=298731
rdar://160392755

Reviewed by Alex Christensen.

Started moving the first 10 functions using postMessageWithAsyncReply to the UI process.

* Tools/WebKitTestRunner/InjectedBundle/Bindings/TestRunner.idl:
* Tools/WebKitTestRunner/InjectedBundle/TestRunner.cpp:
(WTR::TestRunner::displayAndTrackRepaints): Deleted.
(WTR::TestRunner::clearBackForwardList): Deleted.
(WTR::TestRunner::addChromeInputField): Deleted.
(WTR::TestRunner::removeChromeInputField): Deleted.
(WTR::TestRunner::setTextInChromeInputField): Deleted.
(WTR::TestRunner::selectChromeInputField): Deleted.
(WTR::TestRunner::getSelectedTextInChromeInputField): Deleted.
(WTR::TestRunner::focusWebView): Deleted.
(WTR::TestRunner::setBackingScaleFactor): Deleted.
(WTR::TestRunner::removeAllCookies): Deleted.
* Tools/WebKitTestRunner/InjectedBundle/TestRunner.h:
* Tools/WebKitTestRunner/TestController.cpp:
(WTR::if):
(WTR::CompletionHandler<void):

Canonical link: https://commits.webkit.org/300064@main
